src/status_im/contexts/profile/backup_recovery_phrase/view.cljs

```diff
@@ -2,123 +2,167 @@
   (:require
     [clojure.string :as string]
     [native-module.core :as native-module]
-    [quo.context]
+    [quo.context :as quo.context]
     [quo.core :as quo]
     [react-native.core :as rn]
-    [reagent.core :as reagent]
+    [status-im.common.events-helper :as events-helper]
     [status-im.contexts.profile.backup-recovery-phrase.style :as style]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]
     [utils.security.core :as security]))
 
+(def ^:private step-labels
+  [:t/backup-step-1 :t/backup-step-2 :t/backup-step-3 :t/backup-step-4])
+
+(defn- header
+  [{:keys [back-button? shell?]}]
+  [:<>
+   [quo/page-nav
+    (cond-> {:icon-name           (if back-button? :i/arrow-left :i/close)
+             :on-press            events-helper/navigate-back
+             :accessibility-label :top-bar}
+      shell? (assoc :background :blur))]
+   [quo/page-top
+    {:title            (i18n/label :t/backup-recovery-phrase)
+     :description      :text
+     :description-text (i18n/label :t/backup-recovery-phrase-description)}]])
+
 (defn- word-item
-  [item index _ increment]
+  [{:keys [index text]}]
   [rn/view {:style style/word-item}
-   [quo/counter {:type :grey} (+ index increment)]
-   [quo/text {:style {:margin-left 4}} item]])
+   [quo/counter {:type :grey :blur? true}
+    index]
+   [quo/text text]])
+
+(defn words-column
+  [{:keys [starting-index words]}]
+  [rn/view {:style style/words-colum}
+   (map-indexed (fn [idx word]
+                  (let [word-number (+ starting-index idx)]
+                    (with-meta [word-item {:index word-number :text word}]
```

Comment:
Is there a specific reason to prefer `with-meta` instead of the metadata syntax with `^{...}` ?

Reply:
I've seen multiple times the metadata doesn't survive when nested or with complex operations.

So I prefer to use the `^{...}` syntax when the expression is very simple. For this case it works in both ways